src/d/actor/d_a_shand.cpp

```diff
@@ -4,91 +4,488 @@
+    
+    if(actor->mLineMat.init(1, 20, res, 1) == FALSE){
+        return FALSE;
+    }
+    else{
+        return TRUE;
+    }
 }
 
 /* 00002360-00002380       .text daShand_solidHeapCB__FP10fopAc_ac_c */
-static BOOL daShand_solidHeapCB(fopAc_ac_c*) {
-    /* Nonmatching */
+static BOOL daShand_solidHeapCB(fopAc_ac_c* actor) {
+    return useHeapInit(static_cast<shand_class*>(actor));
 }
 
 /* 00002380-00002630       .text daShand_Create__FP10fopAc_ac_c */
-static cPhs_State daShand_Create(fopAc_ac_c*) {
-    /* Nonmatching */
+static cPhs_State daShand_Create(fopAc_ac_c* f_this) {
+    static dCcD_SrcCyl tg_cyl_src = {
+        0,
+        0, 
+        0,
+        0,
+        AT_TYPE_ALL - (AT_TYPE_WATER | AT_TYPE_UNK20000 | AT_TYPE_WIND | AT_TYPE_UNK400000 | AT_TYPE_LIGHT), // 0xff1dfeff 
+        cCcD_TgSPrm_Set_e | cCcD_TgSPrm_IsEnemy_e,
+        cCcD_CoSPrm_Set_e | cCcD_CoSPrm_IsPlayer_e | cCcD_CoSPrm_VsGrpAll_e, 
+        0, 
+        0, 
+        0,
+        0,
+        0,
+        0,
+        0,
+        0,
+        0, 
+        0,
+        0,
+        0,
+        0,
+        0,
+        15.0f, 
+        200.0f
+    };
+
+    static dCcD_SrcSph bm_sph_src = {
+        0,
+        0,
+        0,
+        5,
+        AT_TYPE_ALL - (AT_TYPE_WATER | AT_TYPE_UNK20000 | AT_TYPE_WIND | AT_TYPE_UNK400000 | AT_TYPE_LIGHT), 
+        cCcD_TgSPrm_Set_e | cCcD_TgSPrm_IsEnemy_e,
+    0,
+        0, 
+        0, 
+        0,
+        0,
+        0,
+        0,
+        0,
+        0,
+        0, 
+        0,
+        0,
+        0,
+        0,
+        0,
+        80.0f
+    };
+
+    fopAcM_SetupActor(f_this, shand_class);
+    shand_class* s_this = static_cast<shand_class*>(f_this);
+    int ret = dComIfG_resLoad(&s_this->mPhs, "Shand");
+    if(ret == cPhs_COMPLEATE_e){
+        if(fopAcM_entrySolidHeap(s_this, daShand_solidHeapCB, 0x1040) != false){
+            s_this->health = 2;
+            s_this->mExecuteCount = cM_rndF(10000.0f);
+            s_this->field_318 = 10;
+            s_this->field_2BA = 1;
+            s_this->field_2F0 = 1.0f;
+            if((fopAcM_GetParam(s_this) & 0xff) == true)
+                s_this->field_304 = 15.75f;
+            else
+                s_this->field_304 = 10.5f;
+            s_this->mStts.Init(0xff, 0xff, s_this);
+            for(int i = 0; i < 5; i++){
+                s_this->mCylArr[i].Set(tg_cyl_src);
+                s_this->mCylArr[i].SetStts(&s_this->mStts);
+            }
+            s_this->mSph.Set(bm_sph_src);
+            s_this->mSph.SetStts(&s_this->mStts);
+            s_this->field_2C4 = 30;
+            for(int i = 0; i < 3; i++){
+                daShand_Execute(s_this);
+            }
+        }
+        else {
+            ret = cPhs_ERROR_e;
+        }
+
+        if(hio_set == false){
+            hio_set = true;
+            s_this->mHasHIO = true;
+            l_HIO.mId = mDoHIO_createChild("\x94\xc4\x97\x70\x90\x47\x8e\xe8", &l_HIO);
```

Comment:
Shift JIS text strings should be written out in Japanese characters so they're readable. Ghidra should show you the actual string so you can copy paste it. If Ghidra isn't working for some reason, you can use Python and run `b"\x94\xc4\x97\x70\x90\x47\x8e\xe8".decode("shift-jis")` instead to decode it.